src/fides/api/schemas/privacy_notice.py

```diff
@@ -61,12 +61,38 @@ def validate_data_uses(self, valid_data_uses: List[str]) -> None:
             if data_use not in valid_data_uses:
                 raise ValueError(f"Unknown data_use '{data_use}'")
 
+
+class PrivacyNoticeCreation(PrivacyNotice):
+    """
+    An API representation of a PrivacyNotice.
+    This model doesn't include an `id` so that it can be used for creation.
+    It also establishes some fields _required_ for creation
+    """
+
+    name: str
+    regions: conlist(PrivacyNoticeRegion, min_items=1)  # type: ignore
+    consent_mechanism: ConsentMechanism
+    data_uses: conlist(str, min_items=1)  # type: ignore
+    enforcement_level: EnforcementLevel
+
+    @root_validator(pre=True)
+    def validate_notice_key(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Generate the notice_key from the name if not supplied
+        """
+        if not values.get("notice_key"):
+            values["notice_key"] = PrivacyNoticeModel.generate_notice_key(
+                values.get("name")
+            )
+
+        return values
+
```

Comment:
The diff here looks strange but I mostly moved the `validate_consent_mechanisms_and_display` under PrivacyNoticeCreation instead of the base `PrivacyNotice`